tests/hotplug/memory.go

```diff
@@ -0,0 +1,197 @@
+package hotplug
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"kubevirt.io/kubevirt/tests/libmigration"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/migration"
+
+	util2 "kubevirt.io/kubevirt/tests/util"
+
+	"kubevirt.io/kubevirt/tests/testsuite"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/resource"
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	. "kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/libwait"
+)
+
+var _ = Describe("[sig-compute][Serial]Memory Hotplug", decorators.SigCompute, decorators.SigComputeMigrations, decorators.RequiresTwoSchedulableNodes, decorators.VMLiveUpdateFeaturesGate, Serial, func() {
```

Comment:
These tests look good. 
Would it make sense to try executing both CPU and Memory hot plugs together on the same spec?
What would you expect to happen?

Reply:
So to follow up, in theory, nothing bad should happen because memory hotplug patches the VMI memory requests so that when the memory overhead gets calculated everything should be fine and correct.

Reply:
Added a functional test for this.